Feature/home

minju1009 opened the pull request:
## PR Type

- [ ] 버그 수정
- [x] 신규 기능
- [ ] 기타 (기타인 경우 내용 입력 : )

## 작업 내용
- 홈 화면 MVC 구현
- axios 라이브러리 추가 및 커스텀

## 예상되는 결과
- App 생성시 Home화면 렌더링
- Home 화면 하단의 버튼 클릭시에도 헤더와 동일하게 navigation 가능

## 리뷰어가 중점적으로 봤으면 하는 부분
- MVC 역할에 맞게 잘 구현이 되었는지 

## 예시 화면

https://user-images.githubusercontent.com/90900882/232326615-5804c05d-ceee-49f6-b853-8b3eca250971.mov

minju1009 committed:
feat: 홈 화면 MVC 구현

minju1009 committed:
chore: axios 추가 및 커스텀 axiosClient 추가

minju1009 committed:
style: 홈 화면 스타일 및 템플릿 추가

minju1009 committed:
style: variable에 색상 추가

minju1009 committed:
revert: helper와 View 복구

minju1009 committed:
feat: search 페이지 MVC 구현

minju1009 committed:
style: search 페이지 스타일 적용

minju1009 committed:
feat: search 페이지 template 추가

minju1009 committed:
feat: view component들 추가

minju1009 committed:
feat: randome key 생성하는 util 함수 추가

minju1009 committed:
feat: 최근검색어 삭제기능 추가

minju1009 committed:
refactor: 최근 검색기록에 동일 단어 있을 시 제일 앞에 보여주도록 수정

woowa-boku commented on src/pages/Home/HomeModel.js:
```diff
@@ -0,0 +1,15 @@
+    });
+
+    const gifUrl = data.data.images.original.url;
+    return gifUrl;
```
리턴을 바로하는거라면 변수에 담지 않아도 됩니다.

woowa-boku replied:
아래처럼 사용해도 좋지 않을까 싶네요

```ts
constructor() {
  this.model.init()
}

const data = this.model.toJson()
```

woowa-boku commented on src/pages/View.js:
```diff
@@ -0,0 +1,22 @@
+
+  show(target = this.element) {
+    target.style.display = 'visible';
+    return this;
```
이 케이스에서는 빌더 패턴을 사용하지 않아도 될 것 같아요.

woowa-boku reviewed (approved):
LGTM 👍

snaps-ariel committed:
style: 카드 hover시 선택 효과 적용

snaps-ariel committed:
refactor: 피드백 사항 반영
1. 클래스 필드 대신 명시적 바인딩
2. ?을 이용한 조건문 줄이기
3. data fetching 로직 분리 및data만 return하도록 수정

minju1009 committed:
Merge pull request #1 from minju1009/feature/search3

Feature/search3
